treadmill/supervisor.py

```diff
@@ -241,7 +241,7 @@ def get_state(svcroot):
 
 def _service_wait(svcroot, up_opt, any_all_opt, timeout=0, subset=None):
     """Given services directory, wait for services to be in given state."""
-    services = glob.glob(os.path.join(svcroot, '*'))
+    services = sorted(glob.glob(os.path.join(svcroot, '*')))
```

Comment:
why is this needed?

Reply:
When running the test suite on my Linux system, the test treadmill/tests/supervisor_test.py fails on line 117. The test creates two folders (/a, /b), then checks to see if they are loaded into the _service_wait function in chronological order (/a, /b). During my test run, they are loaded in reverse chronological order (/b, /a). 

I checked the glob module and it states that results are returned in arbitrary order. Because the test assumes results will be ordered, I think that other functions that use the _service_wait function might make that assumption as well. I checked the supervisor_test commit history and I see that the test has been altered recently from testing against a single folder to include testing against multiple folders. 

The sorted() function was the simplest solution I thought of to solve this issue.

Reply:
Ya this code is flaky. It fails our build as well. But in Morgan ms branch supervisor code is updated (refactored) and this goes away. So we are waiting for the new sync (which will happen in a day or two)